File: src/lib/helpers_test.ts
import { assertEquals } from "https://deno.land/std@0.204.0/assert/assert_equals.ts"
import { filterPresentRows } from "./helpers.ts"

Deno.test(function testFilterPresentRows() {
  assertEquals(
    filterPresentRows({
      name: "theName",
      lines: [
        "one",
        "",
        "#comment",
        "three",
        "four",
        "#comment2",
        "five",
      ],
      startLineNr: 10,
    }),
    {
      lines: ["one", "three", "four", "five"],
      indexToLineNr: [10, 13, 14, 16],
    },
  )
})

File: internal/layout/helpers_test.go
package layout

import "testing"

func TestFilterPresentRows(t *testing.T) {
	t.Parallel()

	got := FilterPresentRows(Block{
		Name: "theName",
		Lines: []string{
			"one",
			"",
			"#comment",
			"three",
			"four",
			"#comment2",
			"five",
		},
		StartLineNr: 10,
	})

	want := PresentLines{
		Lines:         []string{"one", "three", "four", "five"},
		IndexToLineNr: []int{10, 13, 14, 16},
	}

	if len(got.Lines) != len(want.Lines) {
		t.Fatalf("expected %d lines, got %d", len(want.Lines), len(got.Lines))
	}
	for i := range want.Lines {
		if got.Lines[i] != want.Lines[i] {
			t.Fatalf("line %d: expected %q, got %q", i, want.Lines[i], got.Lines[i])
		}
	}
	for i := range want.IndexToLineNr {
		if got.IndexToLineNr[i] != want.IndexToLineNr[i] {
			t.Fatalf("line number %d: expected %d, got %d", i, want.IndexToLineNr[i], got.IndexToLineNr[i])
		}
	}
}
